reco-service/src/main/java/com/movie/recommender/reco/service/RecommendationService.java

```diff
@@ -99,11 +104,26 @@ private List<Long> filterPredictions(PredictResponse predictResponse, double thr
                 .collect(Collectors.toList());
     }
 
-    private List<MovieDetails> fetchMovieDetails(List<Long> movieIds) {
-        log.info("Fetching movie details for filtered movie IDs: {}", movieIds);
-        return movieIds.stream()
-                .map(mongoDBService::getMovieDetailsById)  // Fetch movie details from MongoDB
-                .filter(Objects::nonNull)  // Exclude null values if any movie is not found
-                .collect(Collectors.toList());
+    private List<MovieWithShowtime> fetchMoviesWithShowtimes(List<Long> movieIds, LocationDTO location) {
+        log.info("Fetching movies with showtimes for IDs: {}", movieIds);
+
+        List<MovieWithShowtime> moviesWithShowtimes = new ArrayList<>();
+        for (Long movieId : movieIds) {
+            MovieDetails movieDetails = mongoDBService.getMovieDetailsById(movieId);
+
+            // Fetch showtimes using the user's location
+            List<Showtime> showtimes = mongoDBService.getMovieShowtimesByCity(
```

Comment:
calling `getMovieShowtimesByCity` for each movie and then performing a lookup inside is too inefficient and can be optimized
modify `getMovieShowtimesByCity` to return a `Map<Long, List<Showtime>>` where the key is the movieId and the value is the list of showtimes for that movie
with this change you can replace the current logic with the following
```
MovieDetails movieDetails = mongoDBService.getMovieDetailsById(movieId);
List<Showtime> showtimes = showtimesMap.getOrDefault(movieId, Collections.emptyList());
```
this reduces redundant database queries and simplifies the code